File: Code Examples/JWT/java/SecurityExample/src/main/java/io/hashimati/security/domains/RefreshToken.java
package io.hashimati.security.domains;

import io.micronaut.data.annotation.DateCreated;
import io.micronaut.data.annotation.GeneratedValue;
import io.micronaut.data.annotation.Id;
import lombok.Data;

import java.time.Instant;

@Data
public class RefreshToken {
    @Id
    @GeneratedValue(GeneratedValue.Type.AUTO)
    private Long id;
    private String username;
    private String refreshToken;
    private Boolean revoked;
    @DateCreated
    private Instant dateCreated;
}

File: Code Examples/JWT/Kotlin/JWTKotlinJDBC/src/main/kotlin/io/hashimati/security/domains/RefreshToken.kt
package io.hashimati.security.domains

import io.micronaut.data.annotation.*
import io.micronaut.data.model.naming.NamingStrategies
import java.time.Instant
import javax.validation.constraints.NotBlank

@MappedEntity
data class RefreshToken(
    @field:Id
    @GeneratedValue(GeneratedValue.Type.AUTO)
    var id: Long? = null,

    @NotBlank
    var username: String,

    @NotBlank
    var refreshToken: String,

    var revoked: Boolean,

    @DateCreated
    var dateCreated: Instant? = null

)